From 2ca3e4abe7d28aaf98c54210ade8afab0edd0ec8 Mon Sep 17 00:00:00 2001
From: NeilBrown <neilb@suse.de>
Date: Wed, 9 Jul 2014 12:33:15 +1000
Patch-mainline: submitted 2014-07-24
References: bnc#866130
Subject: [PATCH 13/13] autofs4: don't take spinlock when not needed in
 autofs4_lookup_expiring

If the expiring_list is empty, we can avoid a costly spinlock
in the rcu-walk path through authfs4_d_manage.

Signed-off-by: NeilBrown <neilb@suse.de>
---
 fs/autofs4/root.c |    2 ++
 1 file changed, 2 insertions(+)

--- linux-3.0-SLE11-SP3.orig/fs/autofs4/root.c
+++ linux-3.0-SLE11-SP3/fs/autofs4/root.c
@@ -219,6 +219,8 @@ static struct dentry *autofs4_lookup_exp
 	const unsigned char *str = name->name;
 	struct list_head *p, *head;
 
+	if (list_empty(&sbi->expiring_list))
+		return NULL;
 	spin_lock(&sbi->lookup_lock);
 	head = &sbi->expiring_list;
 	list_for_each(p, head) {
